api/helpers/uploadedFileErrorChecks.js

```diff
@@ -0,0 +1,271 @@
+        const fileToDeleteInSession =
+            file.originalname === uploadedFile.filename;
+        if (fileToDeleteInSession) {
+            deleteFileFromStorage(uploadedFile, s3BucketName);
+        }
+        return uploadedFile.filename !== file.originalname;
+    });
+}
+
+function addInfectedFilenameToSessionErrors(req, file) {
+    req.session.eApp.uploadMessages.infectedFiles = [
+        ...req.session.eApp.uploadMessages.infectedFiles,
+        file.originalname,
+    ];
+}
+
+function addCleanAndUnsubmittedTagsToFile(file, req) {
+    const uploadedStorageName = getStorageNameFromSession(file, req);
+    const fileNotInfected = {
+        Key: 'av-status',
+        Value: 'CLEAN',
+    };
+    const restoreUnsubmittedTag = {
+        Key: 'app_status',
+        Value: 'UNSUBMITTED',
+    };
+    const params = {
+        Bucket: req._sails.config.eAppS3Vals.s3_bucket,
+        Key: uploadedStorageName,
+        Tagging: {
+            TagSet: [fileNotInfected, restoreUnsubmittedTag],
+        },
+    };
+
+    s3.putObjectTagging(params, (err) => {
+        if (err) {
+            throw new Error(err);
+        }
+    });
+    sails.log.info(
+        `Both CLEAN and UNSUBMITTED tags added to ${uploadedStorageName}`
+    );
+}
+
+function checkTypeSizeAndDuplication(req, file, cb) {
+    let errors = [];
+    const preventFileUpload = () => cb(null, false);
+    const allowFileUplaod = () => cb(null, true);
+    const { uploadedFileData } = req.session.eApp;
+    const fileAlreadyExists = uploadedFileData.find(
+        (existing) => existing.filename === file.originalname
+    );
+
+    if (file.mimetype !== 'application/pdf') {
+        errors.push(
+            'The file is in the wrong format. Only .pdf files are allowed.'
+        );
+    }
+    if (fileAlreadyExists) {
+        errors.push(
+            'You have already uploaded this file. You cannot upload the same file twice.'
+        );
+    }
+
+    if (errors.length > 0) {
+        addErrorsToSession(req, file, errors);
+        preventFileUpload();
+    } else {
+        allowFileUplaod();
+    }
+}
+
+function addErrorsToSession(req, file, errors) {
+    const fileNamesWithErrors = req.session.eApp.uploadMessages.errors.map(
+        (error) => error.hasOwnProperty('filename') && error.filename
+    );
+    if (fileNamesWithErrors.includes(file.originalname)) {
+        fileNamesWithErrors.forEach((fileName, idx) => {
+            if (fileName === file.originalname) {
+                req.session.eApp.uploadMessages.errors[idx].errors = [
+                    ...req.session.eApp.uploadMessages.errors[idx].errors,
+                    ...errors,
+                ];
+            }
+        });
+    } else {
+        req.session.eApp.uploadMessages.errors.push({
+            filename: file.originalname,
+            errors,
+        });
+    }
+}
+
+function displayErrorAndRemoveLargeFiles(req) {
+    const TWO_HUNDRED_MEGABYTES =
```

Comment:
What if you change the 'file_upload_size_limit' value from 200 to 300? :)

Make change the name of the const TWO_HUNDRED_MEGABYTES to be more general